packages/asset/contracts/Catalyst.sol

```diff
@@ -6,100 +6,141 @@ import "@openzeppelin/contracts-upgradeable/token/ERC1155/ERC1155Upgradeable.sol
 import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC1155/extensions/ERC1155BurnableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC1155/extensions/ERC1155SupplyUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/extensions/ERC1155URIStorageUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "./OperatorFilter/OperatorFiltererUpgradeable.sol";
+import "./RoyaltyHandler.sol";
 import "./ERC2771Handler.sol";
 import "./interfaces/ICatalyst.sol";
 
+/// @title Catalyst
+/// @dev A ERC1155 contract that manages catalysts, extends multiple OpenZeppelin contracts to
+/// provide a variety of features including, AccessControl, URIStorage, Burnable and more.
+/// The contract includes support for meta transactions.
 contract Catalyst is
     ICatalyst,
     Initializable,
     ERC1155Upgradeable,
     ERC1155BurnableUpgradeable,
     ERC1155SupplyUpgradeable,
+    ERC1155URIStorageUpgradeable,
     ERC2771Handler,
-    AccessControlUpgradeable
+    RoyaltyHandler,
+    AccessControlUpgradeable,
+    OperatorFiltererUpgradeable
 {
     bytes32 public constant MINTER_ROLE = keccak256("MINTER");
 
-    uint256 public constant COMMON_CATALYST_ID = 1;
-    uint256 public constant UNCOMMON_CATAYST_ID = 2;
-    uint256 public constant RARE_CATALYST_ID = 3;
-    uint256 public constant EPIC_CATALYST_ID = 4;
-    uint256 public constant LEGENDARY_CATALYST_ID = 5;
-    uint256 public constant MYTHIC_CATALYST_ID = 6;
-
-    uint256 public catalystTypeCount = 6;
-
-    address private royaltyRecipient;
-    mapping(uint256 => uint256) private catalystRoyaltyBps;
+    uint256 public tokenCount;
 
     event TrustedForwarderChanged(address indexed newTrustedForwarderAddress);
     event NewCatalystTypeAdded(uint256 catalystId, uint256 royaltyBps);
 
```

Comment:
missing the constructor to disable init
https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#initializing_the_implementation_contract
```
    constructor() {
        _disableInitializers();
    }
    ```

Reply:
Added!